rules/ast/ast.go

```diff
@@ -29,7 +29,12 @@ import (
 	"github.com/prometheus/prometheus/storage/metric"
 )
 
-var stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+var (
+	stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+	queryTimeout = flag.Duration("query.timeout", 30*time.Second, "Maximum time a query may take before being aborted.")
```

Comment:
In the end this should be user-configurable via a query parameter. But for now, a flag with a default seems ok. Let's set the default to 2 minutes though (based on experience with heavy datasets and some rare, but useful, expensive queries).